game-engine/src/commonMain/kotlin/za/co/entelect/challenge/game/engine/map/WormsMapGenerator.kt

```diff
@@ -84,10 +84,20 @@ class WormsMapGenerator(private val config: GameConfig, private val seed: Int) {
                     val t = 2.0 * PI * (it.toDouble() / count.toDouble()) + tilt
                     val x = radius * cos(t) + mapCenter.first
                     val y = radius * sin(t) + mapCenter.second
-                    getCellAt(Point(x.roundToInt(), y.roundToInt()), blankMap)!!
+                    getCellAt(roundCoordinatesToIntByAngle(t, x, y), blankMap)!!
                 }
     }
 
+    private fun roundCoordinatesToIntByAngle(angle: Double, x: Double, y: Double): Point {
+        return when {
+            angle <= (PI * 0.5) -> Point(ceil(x).toInt(), floor(y).toInt())
+            angle <= (PI * 1.0) -> Point(floor(x).toInt(), floor(y).toInt())
+            angle <= (PI * 1.5) -> Point(floor(x).toInt(), ceil(y).toInt())
+            angle <= (PI * 2.0) -> Point(ceil(x).toInt(), ceil(y).toInt())
+            else -> Point(0, 0)
```

Comment:
should i rather use some kind of `throw exception` here?
This currently won't support `tilt`, since that could make `angle` > `Pi*2`